src/theme/DocVersionBanner/index.tsx

```diff
@@ -0,0 +1,179 @@
+import Translate from '@docusaurus/Translate';
+import { type GlobalVersion } from '@docusaurus/plugin-content-docs/client';
+import { ThemeClassNames } from '@docusaurus/theme-common';
+import { useDocsVersion } from '@docusaurus/theme-common/internal';
+import type { Props } from '@theme/DocVersionBanner';
+import type {
+  VersionBanner,
+  PropVersionMetadata,
+} from '@docusaurus/plugin-content-docs';
+import { useWikiPreferredVersion } from '@site/src/utils/wikiPreferredVersion';
+import { useLocation } from '@docusaurus/router';
+import {
+  useAllLatestVersion,
+  useCurrentDocPlugins,
+} from '@site/src/utils/wikiVersion';
+
+type BannerLabelComponentProps = {
+  siteTitle: string;
+  versionMetadata: PropVersionMetadata;
+};
+
+function UnreleasedVersionLabel({
+  versionMetadata,
+}: BannerLabelComponentProps) {
+  return (
+    <Translate
+      id='theme.docs.versions.unreleasedVersionLabel'
+      description="The label used to tell the user that he's browsing an unreleased doc version"
+      values={{
+        versionLabel: <b>{versionMetadata.label}</b>,
+      }}
+    >
+      {'This is unreleased documentation for {versionLabel} version.'}
+    </Translate>
+  );
+}
+
+function UnmaintainedVersionLabel({
+  versionMetadata,
+}: BannerLabelComponentProps) {
+  return (
+    <Translate
+      id='theme.docs.versions.unmaintainedVersionLabel'
+      description="The label used to tell the user that he's browsing an unmaintained doc version"
+      values={{
+        versionLabel: <b>{versionMetadata.label}</b>,
+      }}
+    >
+      {
+        'This is documentation for {versionLabel}, which is no longer actively maintained.'
+      }
+    </Translate>
+  );
+}
+
+const BannerLabelComponents: {
+  [banner in VersionBanner]: ComponentType<BannerLabelComponentProps>;
+} = {
+  unreleased: UnreleasedVersionLabel,
+  unmaintained: UnmaintainedVersionLabel,
+};
+
+function BannerLabel(props: BannerLabelComponentProps) {
+  const BannerLabelComponent =
+    BannerLabelComponents[props.versionMetadata.banner!];
+  return <BannerLabelComponent {...props} />;
+}
+
+function LatestVersionSuggestionLabel({
+  versionLabel,
+  to,
+  onClick,
+}: {
+  to: string;
+  onClick: () => void;
+  versionLabel: string;
+}) {
+  return (
+    <Translate
+      id='theme.docs.versions.latestVersionSuggestionLabel'
+      description='The label used to tell the user to check the latest version'
+      values={{
+        versionLabel,
+        latestVersionLink: (
+          <b>
+            <Link to={to} onClick={onClick}>
+              <Translate
+                id='theme.docs.versions.latestVersionLinkLabel'
+                description='The label used for the latest version suggestion link label'
+              >
+                latest version
+              </Translate>
+            </Link>
+          </b>
+        ),
+      }}
+    >
+      {
+        'For up-to-date documentation, see the {latestVersionLink} ({versionLabel}).'
```

Comment:
Currently the text feels a bit weird, saying 'up-to-date' pointing to an older version. I'd change it to something like:

```suggestion
        'For the latest stable release, see the {latestVersionLink} ({versionLabel}).'
```

Reply:
It is what docusaurus does. But yeah we can change it